### PR TITLE
refactor(shared, dal, application-generic, api, web, dashboard): Remove `json-schema-to-ts` dependency and use `JsonSchemaDto` everywhere

### PR DESCRIPTION
**Pull request by rifont**
### What changed? Why was the change needed?

* Remove `json-schema-to-ts` dependency and use `@novu/shared`'s `JsonSchemaDto` everywhere

### Screenshots

<details>
<summary><strong>Expand for optional sections</strong></summary>

### Special notes for your reviewer

</details>

### CODE REVIEW

File: packages/shared/src/dto/workflows/json-schema-dto.ts [hunk 1]
```diff
@@ -14,61 +14,61 @@ export type JSONSchemaVersion = string;
 
 export type JSONSchemaDefinition = JSONSchemaDto | boolean;
 
-// eslint-disable-next-line @typescript-eslint/naming-convention
-export interface JSONSchemaDto {
-  type?: JSONSchemaTypeName | JSONSchemaTypeName[] | undefined;
-  enum?: JSONSchemaType[] | undefined;
-  const?: JSONSchemaType | undefined;
+export type JSONSchemaDto = Readonly<{
```
Comment (rifont) [inline packages/shared/src/dto/workflows/json-schema-dto.ts:31]: These changes are for compatibility with `json-schema-to-ts`'s declaration of Json Schema 7 typings.

File: packages/framework/src/types/schema.types.ts [hunk 2]
```diff
@@ -1,7 +1,7 @@
 import type { JSONSchema, FromSchema as JsonSchemaInfer } from 'json-schema-to-ts';
 import zod from 'zod';
 
-export type JsonSchema = JSONSchema;
+export type JsonSchema = Exclude<JSONSchema, boolean>;
```
Comment (rifont) [inline packages/framework/src/types/schema.types.ts:4]: The JSON schema used in Framework is a root-level JSON schema, which can't be a `boolean`. For completeness, the `boolean` value in the library package is present to allow for `additionalProperties: JSONSchema | boolean`, which is a valid JSON schema polymorphism.

File: packages/shared/package.json [hunk 3]
```diff
@@ -38,9 +38,6 @@
     }
   },
   "devDependencies": {
-    "@types/bluebird": "^3.5.24",
-    "@types/jest": "29.5.2",
-    "json-schema-to-ts": "^3.0.0",
```
Comment (rifont) [inline packages/shared/package.json:43]: 🧹

**Review by SokratisVidros — APPROVED** [review]
🧹

**Pull request merged**